Framework/DataHandling/src/LoadMask.cpp

```diff
@@ -39,6 +39,234 @@ using namespace Mantid::Kernel;
+        std::string error =
+            "Range string " + rawstring + " has wrong order of detectors ID!";
+        throw std::invalid_argument(error);
+      }
+      pairs.push_back(intstart);
+      pairs.push_back(intend);
+
+    } else { // 3. Treat singles
+      T itemp = boost::lexical_cast<T>(rawstring);
+      singles.push_back(itemp);
+    }
+  } // ENDFOR i
+}
+
+/*
+* Parse a line in an ISIS mask file string to vector
+* Combination of 5 types of format for unit
+* (1) a (2) a-b (3) a - b (4) a- b (5) a- b
+* separated by space(s)
+* @param  ins    -- input string in ISIS ASCII format
+* @return ranges -- vector of a,b pairs converted from input
+*/
+void parseISISStringToVector(const std::string &ins,
+                             std::vector<Mantid::specnum_t> &ranges) {
+  // 1. Split by space
+  std::vector<string> splitstrings;
+  boost::split(splitstrings, ins, boost::is_any_of(" "),
+               boost::token_compress_on);
+
+  // 2. Replace a-b to a - b, remove a-b and insert a, -, b
+  bool tocontinue = true;
+  size_t index = 0;
+  while (tocontinue) {
+    // a) Determine end of loop.  Note that loop size changes
+    if (index == splitstrings.size() - 1) {
+      tocontinue = false;
+    }
+
+    // b) Need to split?
+    vector<string> temps;
+    boost::split(temps, splitstrings[index], boost::is_any_of("-"),
+                 boost::token_compress_on);
+    if (splitstrings[index].compare("-") == 0 || temps.size() == 1) {
+      // Nothing to split
+      index++;
+    } else if (temps.size() == 2) {
+      // Has a '-' inside.  Delete and Replace
+      temps.insert(temps.begin() + 1, "-");
+      splitstrings.erase(splitstrings.begin() + index);
+      for (size_t ic = 0; ic < 3; ic++) {
+        if (temps[ic].size() > 0) {
+          splitstrings.insert(splitstrings.begin() + index, temps[ic]);
+          index++;
+        }
+      }
+    } else {
+      // Exception
+      std::string err = "String " + splitstrings[index] + " has Too many '-'";
+      throw std::invalid_argument(err);
+    }
+
+    if (index >= splitstrings.size())
+      tocontinue = false;
+
+  } // END WHILE
+
+  // 3. Put to output integer vector
+  tocontinue = true;
+  index = 0;
+  while (tocontinue) {
+    // i)   push to the starting vector
+    ranges.push_back(
+        boost::lexical_cast<Mantid::specnum_t>(splitstrings[index]));
+
+    // ii)  push the ending vector
+    if (index == splitstrings.size() - 1 ||
+        splitstrings[index + 1].compare("-") != 0) {
+      // the next one is not '-'
+      ranges.push_back(
+          boost::lexical_cast<Mantid::specnum_t>(splitstrings[index]));
+      index++;
+    } else {
+      // the next one is '-', thus read '-', next
+      ranges.push_back(
+          boost::lexical_cast<Mantid::specnum_t>(splitstrings[index + 2]));
+      index += 3;
+    }
+
+    if (index >= splitstrings.size())
+      tocontinue = false;
+  } // END-WHILE
+
+  splitstrings.clear();
+}
```

Comment:
Is `clear()` necessary? I would think `splitstrings` will anyway die an all-clearing death on the next line.